### PR TITLE
Add workflow to create releases with SLSA provenance generation

### PR DESCRIPTION
**Pull request by dalg24**
Will publish a release with tar.gz and zip archive artifacts and intoto.jsonl signature file when a tag is pushed to the repo

### CODE REVIEW

File: .github/workflows/releases.yml [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    steps:
+      - name: Download kokkos-${{ github.ref_name }}.zip
+        uses: actions/download-artifact@fa0a91b85d4f404e444e00e005971372dc801d16 # v4.1.8
+        with:
+          name: kokkos-${{ github.ref_name }}.zip
+
+      - name: Download kokkos-${{ github.ref_name }}.tar.gz
+        uses: actions/download-artifact@fa0a91b85d4f404e444e00e005971372dc801d16 # v4.1.8
+        with:
+          name: kokkos-${{ github.ref_name }}.tar.gz
```
Comment (dalg24) [inline .github/workflows/releases.yml:76]: Probably could merge these two steps if we care

File: .github/workflows/releases.yml [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+          if-no-files-found: error
+          retention-days: 5
+
+      - name: Upload source code (tar.gz)
+        uses: actions/upload-artifact@89ef406dd8d7e03cfd12d9e0a4a378f454709029 # v4.3.5
+        with:
+          name: kokkos-${{ github.ref_name }}.tar.gz
+          path: kokkos-${{ github.ref_name }}.tar.gz
+          if-no-files-found: error
+          retention-days: 5
```
Comment (dalg24) [inline .github/workflows/releases.yml:44]: Same here, could probably consolidate into a single step

**Review by masterleinad — APPROVED** [review]
Looks OK to me.

**Review by lucbv — APPROVED** [review]
Good improvement!

**Pull request merged**